### PR TITLE
feat: added base job

### PR DESCRIPTION
**Pull request by olenagerasimova**
part of https://github.com/artipie/artipie/issues/1291

Added base job, which extends quartz job and have useful method to shut down the job. This class can be extended in the adapters to create jobs to process proxy(or new) repositories. 

`QuartzService` class was moved to artipie/artipie and removed from http.

Added class for proxy events with artifact key, repo name and owner login.

Extracted interface from `EventQueue`, such approach allows us to create what ever implementations needed: for tests and for the case, when writing data to db is off and queue should not actually collect some items.

### CODE REVIEW

File: src/main/java/com/artipie/scheduling/EventQueue.java [hunk 1]
```diff
@@ -4,61 +4,130 @@
...
 /**
- * Events queue with {@link ConcurrentLinkedQueue} under the hood.
+ * Events queue.
  * Instance of this class can be passed where necessary (to the adapters for example)
  * to add data for processing into the queue.
  * @param <T> Queue item parameter type.
  * @since 1.3
  */
-public final class EventQueue<T> {
+public interface EventQueue<T> {
```
Comment (dgarus) [inline src/main/java/com/artipie/scheduling/EventQueue.java:20]: Why do we need this interface? Can we use pure `Queue<T>`?

**Review by dgarus — CHANGES_REQUESTED** [review]
Thanks!
Please look at my comment.

File: src/main/java/com/artipie/scheduling/EventQueue.java (hunk 1 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/scheduling/EventQueue.java:20]: we can, but with our own interface we can hide under it's hood anything we want, even not a queue, also we can extend it how we want and need. 
Also, if switch to queue now, we will need to correct a lot of adapter again.

**Comment by olenagerasimova** [comment]
@dgarus thanks, removed queue interface. As for test queue implementation: we can simply use java `LinkedQueue`, no other implementations required

**Review by dgarus — APPROVED** [review]
Thanks!
lgtm

**Pull request merged**